internal/controller/processor.go

```diff
@@ -332,3 +354,28 @@ func (processor *AppConfigurationProviderProcessor) calculateRequeueAfterInterva
 
 	return requeueAfterInterval
 }
+
+func checkSecretVersion(existingSecretReferences map[string]*loader.TargetSecretReference, latestSecretReferences map[string]*loader.TargetSecretReference) {
```

Comment:
This method is not well self-explained, check secret version for what?  I would suggest this method has a boolean type return value, and then update the value to `secretReference.UpdateNeeded`